gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -147,14 +145,34 @@ gst_tensor_query_socket_new (query_connection_handle conn_h,
     g_object_unref (resolver);
   }
 
-  *saddr = g_inet_socket_address_new (addr, conn->port);
+  *saddr = g_inet_socket_address_new (addr, port);
   g_object_unref (addr);
 
+  return TRUE;
+}
+
+/**
+ * @brief Create requested socket.
+ */
+static gboolean
+gst_tensor_query_socket_new (query_connection_handle conn_h,
+    GSocketAddress ** saddr)
+{
+  GError *err = NULL;
+  TensorQueryConnection *conn = (TensorQueryConnection *) conn_h;
+
+  if (!gst_tensor_query_get_saddr (conn->host, conn->port, saddr,
+          conn->cancellable)) {
+    nns_loge ("Failed to get socket address");
+    return FALSE;
+  }
+
   /* create sending client socket */
   /** @todo Support UDP protocol */
   conn->socket =
       g_socket_new (g_socket_address_get_family (*saddr), G_SOCKET_TYPE_STREAM,
       G_SOCKET_PROTOCOL_TCP, &err);
+  g_object_unref (*saddr);
```

Comment:
plz check this line. (move this to line 177?)

Reply:
Right. It should not be un-referred here. Updated.